File: apps/server/src/main/java/com/chtholly/tag/service/TagService.java
package com.chtholly.tag.service;

import com.chtholly.tag.api.dto.TagResponse;

import java.util.List;

/** 标签业务：列表查询与帖子标签计数同步。 */
public interface TagService {

    List<TagResponse> listTags(int limit);

    /** 已发布帖子标签变更：对比 old/new 调整 usage_count 并 upsert 标签行。 */
    void syncPublishedPostTags(long creatorId, List<String> oldTags, List<String> newTags);

    /** 已发布帖子删除：递减各标签 usage_count。 */
    void releasePublishedPostTags(List<String> tags);
}

File: apps/web/lib/services/tagService.ts
import { apiFetch } from "./apiClient";
import type { TagItem } from "@/lib/types/tag";

const TAG_PREFIX = "/api/v1/tags";

export const tagService = {
  /** 按引用次数降序返回标签列表 */
  list: (limit = 50) =>
    apiFetch<TagItem[]>(`${TAG_PREFIX}?limit=${limit}`),
};
